array_core/array_core.h

```diff
@@ -0,0 +1,109 @@
+#ifndef ARRAY_CORE_ARRAY_CORE_H_
+#define ARRAY_CORE_ARRAY_CORE_H_
+
+#include <iostream>
+#include <vector>
+#include "plotly_maker/plotly_maker.h"
+#include "utils_core.h"
+
+using std::vector;
+
+
+namespace davis {
+
+enum class visualizationTypes{
+    graph, heatmap, surface
+};
+
+enum class colorscales{
+    gray, rainbow
+};
+
+struct showSettings{
+    visualizationTypes visualType = visualizationTypes::heatmap;
+    colorscales colorscale = colorscales::rainbow;
+};
+
+
+//! two-dimensional vector
+template <typename T>
+bool show(const vector<vector<T>> &data);
+
+//! one-dimensional vector
+template <typename T>
+bool show(const vector<T> &data);
+
+//! two-dimensional array
+template <typename T>
+bool show(T **data, uint64_t arrRows, uint64_t arrCols);
```

Comment:
May be it's better to use **const uint64_t &arrRows** and for all other parameters
which are just for reading